examples/jsm/controls/OrbitControls.js

```diff
@@ -271,7 +286,91 @@ class OrbitControls extends EventDispatcher {
 
 				}
 
+				// adjust camera position
+				let zoomChanged = false;
+				if ( scope.zoomToCursor && performCursorZoom ) {
+
+					let newRadius = null;
+					if ( scope.object.isPerspectiveCamera ) {
+
+						// move the camera down the pointer ray
+						// this method avoids floating point error
+						const prevRadius = offset.length();
+						newRadius = clampDistance( prevRadius * scale );
+
+						const radiusDelta = prevRadius - newRadius;
+						scope.object.position.addScaledVector( dollyDirection, radiusDelta );
+						scope.object.updateMatrixWorld();
+
+					} else if ( scope.object.isOrthographicCamera ) {
+
+						// adjust the ortho camera position based on zoom changes
+						const mouseBefore = new Vector3( mouse.x, mouse.y, 0 );
```

Comment:
@gkjohnson I think `mouseBefore` should have been instantiated once inside the closure.

Likewise, `mouseAfter`.

Reply:
Oops! You're right. That would be a good change.

Reply:
@gkjohnson Something like this.
```js
mouseBefore.set( mouse.x, mouse.y, 0 ).unproject( scope.object );
...
mouseAfter.set( mouse.x, mouse.y, 0 ).unproject( scope.object );
...
scope.object.updateWorldMatrix( false, false ); // if you say it is needed
```

Reply:
I think this is a good change but this PR is already merged and I have other priorities at the moment. I'm happy to review a PR from anyone else to get it in, though.

Reply:
I do not understand the code block, so it will not be me.